test/models/item_test.rb

```diff
@@ -11,6 +11,16 @@ class ItemTest < ActiveSupport::TestCase
     assert item.number
   end
 
+  test "assigns a number, with an existing item" do
+    create(:item, number: 999)
+
+    borrow_policy = create(:borrow_policy)
```

Comment:
I think this test won't fail if we revert the change, we need to explicitly make the code B.